### ISSUE TITLE
Guidelines wrongly document @wit as attribute for <rdgGrp>

### ISSUE THREAD

**Issue by rvdb** [TEIC/TEI#2006]
I think I've spotted some mistakes in the prose of Chapter 12 "Critical Apparatus", concerning the attribute classes of `<rdgGrp>`. The guidelines still document that `@wit` is allowed on `<rdgGrp>`. Yet, this is contradicted by the [specification](https://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-rdgGrp.html) of `<rdgGrp>`. The discussion in issue https://github.com/TEIC/TEI/issues/1004 seems to indicate that `@wit` had been removed, so I guess the prose is out of date in following places:

1.  Section "12.1.2 Readings" states that:
https://github.com/TEIC/TEI/blob/b1a43a47e650cf7ad65a3bb12538d96dac11a3bd/P5/Source/Guidelines/en/TC-CriticalApparatus.xml#L132-L135

Yet, `<rdgGrp>` no longer is member of `att.witnessed`.

2.  Section "12.1.3 Indicating Subvariation in Apparatus Entries" states that:
https://github.com/TEIC/TEI/blob/b1a43a47e650cf7ad65a3bb12538d96dac11a3bd/P5/Source/Guidelines/en/TC-CriticalApparatus.xml#L249-L251

Yet, `@wit` is no longer defined in `att.textCritical`, but has been moved to `att.witnessed`, of which `<rdgGrp>` is no member (so `@wit` is _not_ a valid attribute for `<rdgGrp>`). Additionally, `@resp` and `@cert` now belong to `att.global.responsibility`, which is no longer included in `att.textCritical` (so `@resp` and `@cert` are still valid attributes for `<rdgGrp>`, because it is a member of `att.global.responsibility`).

I'm not sure if these are all the cases; if I find more, I'll post follow-ups in comments.``

**Comment by ralessi** [comment]
If I may pop in this discussion, I would like to mention another one on the same subject [here](https://gitlab.com/ralessi/ekdosis/-/issues/2#note_423110970).  For the record, `ekdosis` is a new [LuaLaTeX package](https://ctan.org/pkg/ekdosis) dedicated to critical editions.  In addition to PDF, `ekdosis` can output a TEI xml compliant file.

**Comment by hcayless** [comment]
@ralessi, I'm glad to know about ekdosis! 

I was initially nervous that I'd done this by accident :-), but as @rvdb notes, it seems to have been done about 8 years ago. One the face of it, I think the dropping of `@wit` from rdgGrp was not really thoroughly considered. But the question is whether to fix the schema or the prose? In the example you give in your link, I'd personally much prefer nested apps to rdgGrps...

**Comment by ralessi** [comment]
Yes, I too much prefer nested apps to rdgGrps.  This is why `\app{}` commands are allowed to nest in the first release of `ekdosis` whereas no such command as `\rdgGrp{}` has been provided.  Having said that, as demonstrated by [the PDF output](https://gitlab.com/ralessi/ekdosis/-/issues/2#note_423110970) that is on Gitlab, rdgGrps may allow for more elegantly written entries in some situations.

So I just introduced a new `\rdgGrp[]{}` command in the development version of ekdosis.  (Actually I am working on this at the time of writing.)  It takes as mandatory argument readings to be grouped and further accepts `type` as optional key-value argument to describe the type of grouping.  Not that I dropped `wit`, but rather that I fail to see the point in specifying beforehand witnesses that still are to be specified as attributes of `<lem>/<rdg>` to be found within `<rdgGrp>`.

Does this mean that the prose, and not the schema, ought to be fixed?  Well, I am still a bit puzzled and wouldn't dare to suggest this.  Interestingly, this intricate business is avoided in the [LDLT-Guidelines](https://digitallatin.github.io/guidelines/LDLT-Guidelines.html#apparatus-criticus-multiple-entries) where it is only advised to use `<note>` to indicate variations on readings!

**Comment by hcayless** [comment]
I feel like if `@wit` on rdgGrp had been available though, we might have recommended this instead for LDLT:
```xml
<app>
  <lem wit="#A">ante A. Gellium</lem>
  <rdgGrp wit="#B #C #D #E">
    <rdg>antea gellium</rdg>
    <note>vel</note>
    <rdg>ante agellium</rdg>
  </rdgGrp>
</app>
```
To explain this a little: the MSS presumably all have ANTEAGELLIUM, and the question is how you read that, with two options presented (_vel_ means _or_). The apparatus should look something like:

ante A. Gellium] _A_ antea gellium _vel_ ante agellium _BCDE_

So I don't like the repetition of the `@wit` on the subsequent rdgs. Might a compromise be that we could put `@wit` back on rdgGrp, but if it is used, then child lem/rdg may not have it?

**Comment by ralessi** [comment]
I totally agree with this way of looking at this.  Yes, I think that if `@wit` had been available on `<rdgGrp>`, then _if used_, it should have been forbidden as attribute of any child `<lem>`/`<rdg>` elements.  And vice versa: to put this in Latin, _vel_ in the apparatus criticus all right, but _aut_ in the schema: viz. `@wit` either in `<rdgGrp>` or in child elements, but not in both places ;)

In my opinion, your example as it is, with the use of `<note>`, is most educational.

If we go this way, then both the prose and the schema are to be fixed.  Did I get this right?

**Comment by hcayless** [comment]
Yes, insofar as we'd need a bit more documentation for the adjusted usage in the Guidelines prose, and obviously, we'd need to add `att.witnessed` to rdgGrp and add a Schematron rule to prevent `@wit` appearing on both rdgGrp and child lem/wit.

**Comment by ralessi** [comment]
I go on with this idea, this time from the point of view of the author of `ekdosis`, bound by the need to produce both an apparatus in print and a valid TEI output from the same items of information.  What if we had `<lem>` and at least one `<rdg>` within `<rdgGrp>`, instead of only `<rdg>`s elements?  To return to your example, what if A had given "ante A. Gellium" and B "ante A. Gelium"?

In an apparatus in print, you have `ante A. Gellium A] ante A. Gelium B antea gellium _vel_ ante agellium CDE`.  That is first the lemma text, then the closest reading, then other readings, either grouped or not.

Now in LaTeX:

````latex
\app{
  \rdgGrp{
    \lem[wit=A]{ante A. Gellium}
    \rdg[wit=B]{ante A. Gelium}
  }
  \rdgGrp[wit={C,D,E}]{
    \rdg{antea gellium}
    \note{vel}
    \rdg{ante agellium}
  }
}
````

Obviously, in the first `<rdgGrp>` you have no other option than to insert `@wit` information in child elements—and rightfully so, as this way the reader knows exactly how each witness reads the passage.  What I mean is this way of encoding always gives fine-grained information as to what is found in every single witness.  And this works in all possible cases.

As to the subsequent `<rdgGrp>`, what did the encoder actually read in D for example?  Is it "antea gellium" or "ante agellium"?  Isn't it rather that the reading of C is dubious?  If so, then wouldn't it be clearer to specify in a `<note>` that such reading is dubious?

To sum up, yes, `@wit` should go either in `<rdgGrp>` or in child elements, not in both. But is there really a point in using it as early as in `<rdgGrp>`?

**Comment by ralessi** [comment]
**[Follow-up]**

Not to mention other inconsistencies?  I wouldn't say for sure, but for instance, readings from previous editions in print are better referred to as sources, with the `@source` attribute.  But since `@source` is global, unlike `@wit`, it is allowed both in `<rdgGrp>` and in child elements.

(I'm just writing this as my thought goes along.  It may be nonsense.)

**[EDIT]**
Speaking of `@wit` in both places, the original authors of the Guidelines wrote the following:
> When values for any of these attributes are given on a `<rdgGrp>` element, the values given are inherited by the `<rdg>` or `<lem>` elements nested within the reading group, unless overridden by a new specification on the individual reading element.

This would allow for two different/subsequent kinds of encoding, by 1. Specifying the witnesses involved in the grouping without going into detail. 2. Giving details in the child elements.

This makes sense, but would a machine need 1. to understand the grouping in case all information is found in child elements by means of `@wit`?  To put it differently: wouldn't 2. be required at any rate?

**Comment by hcayless** [comment]
To be honest, I'm mostly not a fan of rdgGrp at all. I'd be tempted to use it in the example above, and it's useful in the case of variorum editions (where you're recording the apparatus of multiple prior editions). One might be tempted to think you'd want to group types of reading (orthographic variants, lexical variants, etc.) but usually, variants in an apparatus aren't organized like that, but in (the editor's) priority order, so they don't necessarily clump together in groupable ways. I'd only use them in a very small set of circumstances:

1. I'm doing a variorum edition and each rdgGrp has a `@source` indicating the edition it comes from.
2. I want to provide a single `@source` (or, if we put it back, `@wit`) for a group of readings (e.g. a previous editor gives two or more conjectures). Again though, rdgGrp won't help if I decide conjecture 1 is right and I want to make it the lemma, and conjecture 2 is interesting but wrong, so I put it last in the list of rdgs.

I mustn't forget that my own prejudices aren't rules though :-), so editors may want to group their readings in all sorts of mad ways, and I wouldn't want to stop them.

**Comment by ralessi** [comment]
On my side, I must confess that I never used rdgGrp at all in my work!  This may be the reason why I am puzzled when it comes to find a proper way of using it...  I get your points and agree with you, especially about organizing variants in the editor's priority order.  Agreed on both points.

What I don't get (I humbly plead guilty to this):  if we admit that `<lem>/<rdg>` elements must have at least one `@wit` (manuscript) or `@source` (edition) or `@resp` (scholar, individual) attribute to avoid ambiguity, then why would it be needed to replicate that information in `<rdgGrp>`? Why not limit the role of this element to just one thing, namely do the grouping and possibly mention the reason for the grouping with `@type` for example?

I hope I don't sound obtuse on an otherwise obvious point!

**Comment by ebeshero** [comment]
@ralessi @hcayless I'm on a variorum project working with ms and print sources that's using `<rdgGrp>`, so I figured I should chime in here. For what it's worth, reviewing our project code, I know we wouldn't want to have an attribute on `<rdgGrp>` to point to witnesses because we're doing that on the `<rdg>` elements inside. For us the `<rdgGrp>` element is a helpful container, a way for us to bundle together witnesses that share the same content at a specific location.

**Comment by ralessi** [comment]
@ebeshero Precisely.  If I may add one more reason to do as you do, I think that mentioning say ABC as witnesses in `<rdgGrp>` for witnesses that exhibit similar but not identical readings, then encoded in sequence within the group is somewhat confusing. (At least to me.)

**Comment by ebeshero** [comment]
@ralessi Yes, exactly. We're bundling witnesses together into rdgGrps based on their sharing a set of normalized tokens, but that doesn't mean the presentation within each source is always exactly identical.

**Comment by hcayless** [comment]
I think this has wandered off course a little. I see 2 ways of proceeding:

1. Remove the language indicating that `@wit` can appear on rdgGrp.
2. Put `@wit` back on rdgGrp, but add a rule that if it appears there, its child readings can't have `@wit` and add some more prose around how to use it.

`#2` would let me solve an edge case, but I can think of other ways to do it, so I think I'm in favor of `#1`. @ralessi is correct that there's an apparent imbalance in the availability of `@source` on both rdgGrp and rdg/lem, but I think you absolutely need `@source` on rdgGrp for variorum editions and I wouldn't want to remove it.

**Comment by ralessi** [comment]
I, too, am in favor of `#1` which has the added benefit of being the conservative solution.  After all and furthermore, this issue is about a possible inconsistency between the prose and the schema, not about some missing feature.  About `@source` being needed in variorum editions: it certainly is and I missed this point.

I would like to express my grateful thanks to @ebeshero and @hcayless for their useful comments.  If I may, I attach [here](https://gitlab.com/ralessi/ekdosis/-/issues/2#note_425569607) the relevant pages of the documentation of the next release of `ekdosis` about subvariation.  As it stands, the Lua function that is behind the LaTeX command `\rdgGrp{}` is able to handle any other options.  Do tell me if you think that I missed something of importance at this point, knowing that optional arguments can be added very easily.

**Comment by martinascholger** [comment]
Council at VF2F agrees with @hcayless and @ralessi to fix the prose rather than reintroduce `@wit` on `<rdgGrp>`.

**Comment by ralessi** [comment]
Thank you, much interesting discussion!
